I've not had good luck with running most of Marlin's scripts, so I usually send them off to GitHub and let CI handled them.

I still have the same `Permission denied` error when trying to run `make tests-single-local` / `make tests-all-local` unless I run `chmod +x get_test_targets.py` before they'll attempt to run. Could all of these scripts be set to executable by default like we do for files in `Marlin/buildroot/tests/`? Or somehow automate it? Switching branches / testing out all kind of forks makes this a pain.

Once I make `get_test_targets.py` executable and run `make tests-single-local TEST_TARGET=FYSETC_F6`, it has several `macros.h` warnings, but works:
```
Passed
All tests completed successfully
```

Running `make tests-all-local`, it fails with a new error (same error I get before this PR):
```
Traceback (most recent call last):
  File "/Marlin/buildroot/share/scripts/get_test_targets.py", line 11, in <module>
    with open(yaml_file) as f:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '.github/workflows/test-builds.yml'
```

I documented some of these issues & potential fixes in https://github.com/MarlinFirmware/Marlin/pull/26904#issuecomment-2027326052, but it would be *really* handy if there was a set of instructions somewhere that had the list of prerequisites on what all needs to be installed / set to executable or automate it so this process is easier.

--- a/Makefile
+++ b/Makefile
@@ -55,9 +55,10 @@ tests-single-local-docker:
 	$(CONTAINER_RT_BIN) run $(CONTAINER_RT_OPTS) $(CONTAINER_IMAGE) make tests-single-local TEST_TARGET=$(TEST_TARGET) VERBOSE_PLATFORMIO=$(VERBOSE_PLATFORMIO) GIT_RESET_HARD=$(GIT_RESET_HARD) ONLY_TEST="$(ONLY_TEST)"
 
 tests-all-local:
+	@python -c "import yaml" 2>/dev/null || (echo 'pyyaml module is not installed. Install it with "python -m pip install pyyaml"' && exit 1)
 	export PATH="./buildroot/bin/:./buildroot/tests/:${PATH}" \
 	  && export VERBOSE_PLATFORMIO=$(VERBOSE_PLATFORMIO) \
-	  && for TEST_TARGET in $$($(SCRIPTS_DIR)/get_test_targets.py) ; do \
+	  && for TEST_TARGET in $$(python $(SCRIPTS_DIR)/get_test_targets.py) ; do \
 	    echo "Running tests for $$TEST_TARGET" ; \
 	    run_tests . $$TEST_TARGET || exit 1 ; \
 	    sleep 5; \

--- a/Makefile
+++ b/Makefile
@@ -59,6 +59,10 @@ tests-all-local:
 	export PATH="./buildroot/bin/:./buildroot/tests/:${PATH}" \
 	  && export VERBOSE_PLATFORMIO=$(VERBOSE_PLATFORMIO) \
 	  && for TEST_TARGET in $$(python $(SCRIPTS_DIR)/get_test_targets.py) ; do \
+	    if [ "$$TEST_TARGET" = "linux_native" ] && [ "$$(uname)" = "Darwin" ]; then \
+	      echo "Skipping tests for $$TEST_TARGET on macOS" ; \
+	      continue ; \
+	    fi ; \
 	    echo "Running tests for $$TEST_TARGET" ; \
 	    run_tests . $$TEST_TARGET || exit 1 ; \
 	    sleep 5; \

--- a/buildroot/share/scripts/get_test_targets.py
+++ b/buildroot/share/scripts/get_test_targets.py
@@ -5,7 +5,7 @@ Extract the builds used in Github CI, so that we can run them locally
 import yaml
 
 # Set the yaml file to parse
-yaml_file = '.github/workflows/test-builds.yml'
+yaml_file = '.github/workflows/ci-build-tests.yml'
 
 # Parse the yaml file, and load it into a dictionary (github_configuration)
 with open(yaml_file) as f:

